ivadomed/transforms.py

```diff
@@ -181,11 +181,15 @@ def __init__(self, hspace, wspace, dspace=1., interpolation_order=2):
     @multichannel_capable
     @two_dim_compatible
     def undo_transform(self, sample, metadata):
-        assert "resample" in metadata
+        assert "data_shape" in metadata
+        is_2d = sample.shape[-1] == 1
 
         # Get params
-        params_do = metadata["resample"]
-        params_undo = [1. / x for x in params_do]
+        original_shape = metadata["data_shape"]
+        current_shape = sample.shape
+        params_undo = [x / y for x, y in zip(original_shape, current_shape)]
```

Comment:
Works too. Did you have issues with the other version? curious

Reply:
I also wanted to comment on this! I changed it because using the inverse was not enough precise so I didn't have the same input and output shape after doing and undoing transforms. (e.i 51 instead of 52). When using the dimensions, I get a zoom of example 2.01343 instead of 2, which always give me the right output size.

Reply:
Excellent :)